### PR TITLE
Orchestrator Selection and Reliability Strategy Documentation

### PR DESCRIPTION
**Pull request by angyangie**
**What does this pull request do? Explain your changes. (required)**

Documentation for how the O selection mechanism works, the overall reliability strategy, and the justifications for that strategy.

**Specific updates (required)**

The document describes:
- Reliability and load control: round robin, retries
- Backpressure: One segment in flight
- BroadcasterManager list refresh intervals #806 (comment)
- Front running prevention: External storage prefix for multi-O
- Justification for each of these mechanisms, the problem(s) they solve and how

**Does this pull request close any open issues?**
https://github.com/livepeer/go-livepeer/issues/868

**Checklist:**
- [x] README and other documentation updated
- [ ] Node runs in OSX and devenv
- [ ] All tests in `./test.sh` pass

### CODE REVIEW

File: doc/reliability.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+## Orchestrator Selection
+
+To give preference to O's that respond with transcoded segments quickly, instead of selecting an Orchestrator from the beginning of `sessList` when needed, and placing new Orchestrators that are finished processing a segment at the end, `selectSession` takes Orchestrators from the end of `sessList`. If transcoding is successful, it adds them back to the end of `sessList`. 
+
+## Transcoding Errors & Retries
+
+If there is an error uploading segment to an Orchestrator's OS, submitting the segment to an Orchestrator, downloading transcoded segments, or the segment signature check fails, the Orchestrator is removed from the `sessMap`. The segment is retried with a different Orchestrator. When `selectSession` is called in this retry scenario, though the removed session might still exist in `sessList`, only a session that still exists in `sessMap` will be selected.  If there is no error in segment transcoding, `completeSession` adds session back to `sessList`.
```
Comment (darkdarkdragon) [inline doc/reliability.md]: I think it will be good to add that retries stops when `sessMap` is empty.

**Review by darkdarkdragon — CHANGES_REQUESTED** [review]

File: doc/reliability.md (hunk 1 above)
Comment (angyangie) [reply doc/reliability.md]: I'll add that!

**Review by darkdarkdragon — APPROVED** [review]
LGTM

**Comment by j0sh** [comment]
This is a nice start, but it feels a bit heavy on the details of the implementation. 
Referring to particular variables in external documentation is a bit too brittle and specific (eg, `sessMap`, `sessList`); but it's a good indicator of where we can improve the in-code comments.

A higher level description of the behavior would be more accessible, and we can point interested readers to certain areas of the code for further follow-up on their own. Developers should be able to follow the code from the high-level description, while the current description will be confusing to non-developers (users) who might just want to understand the O selection mechanic.

**Comment by angyangie** [comment]
Thanks for the feedback @j0sh. So you think it would be more readable if I replaced `sessList` with sessions list, `HTTPTimeout` with http timeout, `sessMap` with sessions map, `BroadcastSession` with broadcast session? Or you're saying I should re-write the content of a lot of this to simplify?

**Comment by angyangie** [comment]
I will merge this as is, and make edits once @j0sh is back!

**Pull request merged**